joomla/joomla-cms
# [4.0] Null date corrections for frontent edit icon

Pull Request for Issue #26783 , see also [https://github.com/joomla/joomla-cms/pull/26751#issuecomment-544600907](https://github.com/joomla/joomla-cms/pull/26751#issuecomment-544600907) .

### Summary of Changes

Changes the code to produce the edit icon for frontend editing so it handles real null values for published up and down times.

This fixes the issue that on current 4.0-dev, they eye icon is shown for published articles instead of the edit icon, like it would be with unpublished or not published yet articles.

### Testing Instructions

Use an installation of current 4.0-dev or latest nightly build of today.

1. Create an article with state = published and show it on frontend.
2. On frontend, login as used who has permission for frontend editing.
3. Click the edit button (gear icon at the right top corner of article).
4. In the pulldown which appears below the gear icon, check the icon shown for the "Edit" menu item.

Result: See section "Actual result" below.

5. Apply patch.
6. Reload the page on frontend.
7. Check again the icon in the frontend editing pulldown menu.

Result: See section "Expected result" below.

8. Play around with publish up and down times so you have following situations, and check the icon again:

- Published down in past = article expired
- Published up in future = article pending (i.e. not published yet)
- Published up in past, published down in future = article published
- If you want play with other combinations with empty or not empty published up or down fields.

### Expected result

For published articles the edit icon is shown and for unpublished, pending or expired articles the eye icon.

![snap-14](https://user-images.githubusercontent.com/7413183/67241938-bc97b580-f454-11e9-8b25-af7bd7243b20.png)

### Actual result

For published articles the same eye icon is shown as for unpublished, expired or pending articles.

![snap-13](https://user-images.githubusercontent.com/7413183/67241917-b1448a00-f454-11e9-82e6-720df4d65ef7.png)

### Documentation Changes Required

None.
